modules/adlooxAnalyticsAdapter.js

```diff
@@ -0,0 +1,289 @@
+      code = code.replace(/^\d/, '\\3$& ');
+      return `#${code}`
+    },
+    client: config.options.client,
+    clientid: config.options.clientid,
+    tagid: config.options.tagid,
+    platformid: config.options.platformid,
+    params: []
+  };
+
+  config.options.params = utils.mergeDeep({}, PARAMS_DEFAULT, config.options.params || {});
+  Object
+    .keys(config.options.params)
+    .forEach(k => {
+      if (!Array.isArray(config.options.params[k])) {
+        config.options.params[k] = [ config.options.params[k] ];
+      }
+      config.options.params[k].forEach(v => analyticsAdapter.context.params.push([ k, v ]));
+    });
+
+  Object.keys(COMMAND_QUEUE).forEach(commandProcess);
+
+  analyticsAdapter.originEnableAnalytics(config);
+}
+
+analyticsAdapter.originDisableAnalytics = analyticsAdapter.disableAnalytics;
+analyticsAdapter.disableAnalytics = function() {
+  analyticsAdapter.context = null;
+
+  analyticsAdapter.originDisableAnalytics();
+}
+
+analyticsAdapter.url = function(url, args, bid) {
+  // utils.formatQS outputs PHP encoded querystrings... (╯°□°)╯ ┻━┻
+  function a2qs(a) {
+    // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent
+    function fixedEncodeURIComponent(str) {
+      return encodeURIComponent(str).replace(/[!'()*]/g, function(c) {
+        return '%' + c.charCodeAt(0).toString(16);
+      });
+    }
+
+    const args = [];
+    let n = a.length;
+    while (n-- > 0) {
+      if (!(a[n][1] === undefined || a[n][1] === null || a[n][1] === false)) {
+        args.unshift(fixedEncodeURIComponent(a[n][0]) + (a[n][1] !== true ? ('=' + fixedEncodeURIComponent(a[n][1])) : ''));
+      }
+    }
+
+    return args.join('&');
+  }
+
+  const macros = (str) => {
+    return str.replace(/%%([a-z]+)%%/gi, (match, p1) => MACRO[p1] ? MACRO[p1](bid, analyticsAdapter.context) : match);
+  };
+
+  url = macros(url);
+  args = args || [];
+
+  let n = args.length;
+  while (n-- > 0) {
+    if (utils.isFn(args[n][1])) {
+      try {
+        args[n][1] = args[n][1](bid);
+      } catch (_) {
+        utils.logError(MODULE, 'macro', args[n][0], _.message);
+        args[n][1] = `ERROR: ${_.message}`;
+      }
+    }
+    if (utils.isStr(args[n][1])) {
+      args[n][1] = macros(args[n][1]);
+    }
+  }
+
+  return url + a2qs(args);
+}
+
+analyticsAdapter[`handle_${EVENTS.AUCTION_END}`] = function(auctionDetails) {
+  if (!(auctionDetails.auctionStatus == AUCTION_COMPLETED && auctionDetails.bidsReceived.length > 0)) return;
+  analyticsAdapter[`handle_${EVENTS.AUCTION_END}`] = NOOP;
+
+  utils.logMessage(MODULE, 'preloading verification JS');
+
+  const uri = utils.parseUrl(analyticsAdapter.url(`${analyticsAdapter.context.js}#`));
+
+  const link = document.createElement('link');
+  link.setAttribute('href', `${uri.protocol}://${uri.host}${uri.pathname}`);
+  link.setAttribute('rel', 'preload');
+  link.setAttribute('as', 'script');
+  utils.insertElement(link);
```

Comment:
@jimdigriz 

Part of the "policy" of analytics adapters has been "Just use the src/ajax.js lib to signal your event data"

As well as, we do not really allow inserting scripts / elements.

After discussing with the "Rules Committee" we decided that we will allow these kinds of things, since Pubs usually only pick one or two analytics adapters and usually do their due diligence in researching the things they do.

Part of this exception is we have now split out the Analytics Modules in the Prebid Docs repo to their own separate files.

So the ask here is for you to submit a corresponding doc and detail / not some of the different things your adapter does!

Here is the PR which broke out Analytics Adapters: https://github.com/prebid/prebid.github.io/pull/2727

Reply:
Submitted in https://github.com/prebid/Prebid.js/pull/6308#discussion_r589585261

Let me know if I have missed anything else blocking this from moving forward. Thanks.

Reply:
@jimdigriz could you duplicate all the prominent disclosures in your MD files back to this docs pr?

Reply:
done